aten/src/ATen/core/SymInt.h

```diff
@@ -0,0 +1,50 @@
+#include <c10/macros/Macros.h>
+#include <c10/util/Exception.h>
+
+namespace c10 {
+
+// SymInt is a C++ wrapper class used by LTC and AOTAutograd for tracing
+// arithmetic operations on symbolic integers (e.g. sizes).
+//
+// `SymInt` is also a data type in Pytorch that can be used in function schemas
+// to enable tracing.
+//
+// To trace the operations, SymInt overloads arithmetic operators (e.g. +, -, *)
+// and provides overloads taking SymInt for commonly used math functions.
+//
+// SymInt is semantically a union structure Union[int64_t, SymbolicIntNode*]
+// implemented as a single packed int64_t field named data_.
+//
+// data_ can be either a plain int64_t or (1 << 63 | `index`). `index` points to
+// SymbolicIntNode* that is responsible for constructing an IR node for
+// a traced operation to represent it in LTC or Fx graphs.
```

Comment:
It's not important to smith the words too much since you're going to be updating them, but it's good to have descriptions that describe the code accurately as is, and not the hypothetical future (which you can also describe, but be clear about it ;)